multicluster/controllers/multicluster/common/common_area.go

```diff
@@ -21,9 +21,9 @@ import (
 )
 
 // CommonArea is an interface that provides Access to the common area of a cluster.
-// For CommonArea used in LocalClusterManager - this provides access to the local cluster in a given Namespace.
-//   This is used in the member cluster to access local Namespace.
-// For CommonArea used in RemoteCluster - this provides access to the RemoteCluster Namespace.
+// For CommonArea used in LocalCommonArea - this provides access to the local cluster.
+//   This is used in the member cluster to access local cluster.
```

Comment:
I thought I understood but probably I did not. Could you explain again what is LocalCommonArea for? I thought CommonArea is just the Namespace in Leader Cluster to export resources.

Reply:
I think you are right. The original intention was for this to be used in the leader cluster to write to its local namespace.

But since reconciler in leader cluster are the ones writing to local namespace and they already have access to the local cluster client, this was never used.

And it ended up being used in the reconciler running in member controller for resources in Leader's CommonArea, because they only have leader cluster client and dont have the local client, so this LocalCommonArea is used by those reconcilers to write to the member cluster.

So ideally I should remove this abstraction and just pass in a local cluster client to those reconcilers. Can I do this in a follow up PR, because it causes lot of conflicts I think to address in all the outstanding  commits.

Reply:
Sure, can I do it in a follow-up PR to avoid rebasing lot of commits?

Reply:
Sounds good. But can you add the comments in this PR?

Reply:
Sure, I just updated the PR now. If you think everything looks good, I will squash the changes to prepare for the final merge.

Reply:
Ref: 

https://github.com/antrea-io/antrea/pull/3041/commits/05009071f50462f8338e28abdd67066c028275f7#diff-df19905d6ec88b406e6bd641fc1174690aec0150f0e9bc46cd2cf7460592bd9e 

https://github.com/antrea-io/antrea/pull/3041/commits/05009071f50462f8338e28abdd67066c028275f7#diff-ae7a3b761ad419608bc3d570503c19210746b302941e377821089a6258950fc8